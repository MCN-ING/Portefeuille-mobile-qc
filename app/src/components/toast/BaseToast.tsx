import { GenericFn, testIdWithKey, useTheme } from '@hyperledger/aries-bifold-core'
import React from 'react'
import { useTranslation } from 'react-i18next'
import { View, Text, useWindowDimensions, StyleSheet, TouchableOpacity } from 'react-native'
import Toast from 'react-native-toast-message'
import Icon from 'react-native-vector-icons/MaterialIcons'

interface BaseToastProps {
  title?: string
  body?: string
  toastType: string
  onPress?: GenericFn
  onShow?: GenericFn
  onHide?: GenericFn
  onCancel?: GenericFn
}

export enum ToastType {
  Success = 'success',
  Info = 'info',
  Warn = 'warn',
  Error = 'error',
}

const BaseToast: React.FC<BaseToastProps> = ({ title, body, toastType, onPress = () => null, onCancel }) => {
  const { TextTheme, borderRadius, ColorPallet } = useTheme()
  const { t } = useTranslation()
  const { width } = useWindowDimensions()
  const iconSize = 24
  let backgroundColor = ''
  let borderColor = ''
  const iconColor = ColorPallet.grayscale.white
  const textColor = ColorPallet.grayscale.white
  const styles = StyleSheet.create({
    container: {
      flex: 1,
      flexDirection: 'row',
      alignItems: 'center',
      borderRadius,
      padding: 16,
    },
    textContainer: {
      flexShrink: 1,
    },
    title: {
      fontSize: 14,
      fontWeight: TextTheme.bold.fontWeight,
    },
  })
  switch (toastType) {
    case ToastType.Success:
      backgroundColor = ColorPallet.notification.success
      borderColor = ColorPallet.notification.successBorder
      break

    case ToastType.Info:
      backgroundColor = ColorPallet.brand.primary
      borderColor = ColorPallet.brand.primary
      break

    case ToastType.Warn:
      backgroundColor = ColorPallet.notification.warn
      borderColor = ColorPallet.notification.warnBorder
      break

    case ToastType.Error:
      backgroundColor = ColorPallet.semantic.error
      borderColor = ColorPallet.semantic.error
      break

    default:
      throw new Error('ToastType was not set correctly.')
  }

  return (
    <TouchableOpacity activeOpacity={1} onPress={() => onPress()}>
      <View style={[styles.container, { backgroundColor, borderColor, width: width - width * 0.1 }]}>
        <View style={{ flex: 2, flexDirection: 'row' }}>
          <View style={styles.textContainer}>
            <Text style={[TextTheme.normal, styles.title, { color: textColor }]} testID={testIdWithKey('ToastTitle')}>
              {title}
            </Text>
            {body && (
              <Text style={[TextTheme.normal, { color: textColor }]} testID={testIdWithKey('ToastBody')}>
                {body}
              </Text>
            )}
          </View>
        </View>
        <View style={{ flex: 1 }}>
          <TouchableOpacity
            style={{ alignSelf: 'flex-end' }}
            onPress={() => {
              if (typeof onCancel !== 'undefined') onCancel()
              Toast.hide()
            }}
          >
            {typeof onCancel !== 'undefined' ? (
              <Text style={[TextTheme.labelTitle, { color: textColor }]}>{t('Global.Cancel')}</Text>
            ) : (
              <Icon name={'close'} color={iconColor} size={iconSize} />
            )}
          </TouchableOpacity>
        </View>
      </View>
    </TouchableOpacity>
  )
}

export default BaseToast
