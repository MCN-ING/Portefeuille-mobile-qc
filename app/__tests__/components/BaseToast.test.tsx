import { render } from '@testing-library/react-native'
import React from 'react'

import BaseToast, { ToastType } from '../../src/components/toast/BaseToast'

describe('BaseToast Component', () => {
  test('Info renders correctly', () => {
    const tree = render(<BaseToast title={'Any Info Title'} body={'Any info body'} toastType={ToastType.Info} />)

    expect(tree).toMatchSnapshot()
  })

  test('Success renders correctly', () => {
    const tree = render(
      <BaseToast title={'Any Success Title'} body={'Any success message'} toastType={ToastType.Success} />
    )

    expect(tree).toMatchSnapshot()
  })

  test('Warn renders correctly', () => {
    const tree = render(
      <BaseToast title={'Any Warning Title'} body={'Any warning message'} toastType={ToastType.Warn} />
    )

    expect(tree).toMatchSnapshot()
  })

  test('Error renders correctly', () => {
    const tree = render(<BaseToast title={'Any Error Title'} body={'Any error message'} toastType={ToastType.Error} />)

    expect(tree).toMatchSnapshot()
  })

  test('Toast Renders without body text', () => {
    const tree = render(<BaseToast title={'Any Toast Title'} toastType={ToastType.Error} />)
    const bodyText = tree.queryByTestId('ToastBody')
    expect(bodyText).toBeNull()
  })
})
