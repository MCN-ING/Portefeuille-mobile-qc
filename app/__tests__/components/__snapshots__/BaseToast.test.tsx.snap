// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseToast Component Error renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "borderRadius": 4,
          "flex": 1,
          "flexDirection": "row",
          "marginTop": 60,
          "padding": 16,
        },
        {
          "backgroundColor": "#D8292F",
          "borderColor": "#D8292F",
          "width": 675,
        },
      ]
    }
  >
    <View
      style={
        {
          "flex": 2,
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          {
            "flexShrink": 1,
          }
        }
      >
        <Text
          style={
            [
              {
                "color": "#FFFFFF",
                "fontSize": 18,
                "fontWeight": "normal",
              },
              {
                "fontSize": 14,
                "fontWeight": "bold",
              },
              {
                "color": "#FFFFFF",
              },
            ]
          }
          testID="com.ariesbifold:id/ToastTitle"
        >
          Any Error Title
        </Text>
        <Text
          style={
            [
              {
                "color": "#FFFFFF",
                "fontSize": 18,
                "fontWeight": "normal",
              },
              {
                "color": "#FFFFFF",
              },
            ]
          }
          testID="com.ariesbifold:id/ToastBody"
        >
          Any error message
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "flex": 1,
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignSelf": "flex-end",
            "opacity": 1,
          }
        }
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            [
              {
                "color": "#FFFFFF",
                "fontSize": 24,
              },
              undefined,
              {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
        >
          
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`BaseToast Component Info renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "borderRadius": 4,
          "flex": 1,
          "flexDirection": "row",
          "marginTop": 60,
          "padding": 16,
        },
        {
          "backgroundColor": "#42803E",
          "borderColor": "#42803E",
          "width": 675,
        },
      ]
    }
  >
    <View
      style={
        {
          "flex": 2,
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          {
            "flexShrink": 1,
          }
        }
      >
        <Text
          style={
            [
              {
                "color": "#FFFFFF",
                "fontSize": 18,
                "fontWeight": "normal",
              },
              {
                "fontSize": 14,
                "fontWeight": "bold",
              },
              {
                "color": "#FFFFFF",
              },
            ]
          }
          testID="com.ariesbifold:id/ToastTitle"
        >
          Any Info Title
        </Text>
        <Text
          style={
            [
              {
                "color": "#FFFFFF",
                "fontSize": 18,
                "fontWeight": "normal",
              },
              {
                "color": "#FFFFFF",
              },
            ]
          }
          testID="com.ariesbifold:id/ToastBody"
        >
          Any info body
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "flex": 1,
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignSelf": "flex-end",
            "opacity": 1,
          }
        }
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            [
              {
                "color": "#FFFFFF",
                "fontSize": 24,
              },
              undefined,
              {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
        >
          
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`BaseToast Component Success renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "borderRadius": 4,
          "flex": 1,
          "flexDirection": "row",
          "marginTop": 60,
          "padding": 16,
        },
        {
          "backgroundColor": "#313132",
          "borderColor": "#2E8540",
          "width": 675,
        },
      ]
    }
  >
    <View
      style={
        {
          "flex": 2,
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          {
            "flexShrink": 1,
          }
        }
      >
        <Text
          style={
            [
              {
                "color": "#FFFFFF",
                "fontSize": 18,
                "fontWeight": "normal",
              },
              {
                "fontSize": 14,
                "fontWeight": "bold",
              },
              {
                "color": "#FFFFFF",
              },
            ]
          }
          testID="com.ariesbifold:id/ToastTitle"
        >
          Any Success Title
        </Text>
        <Text
          style={
            [
              {
                "color": "#FFFFFF",
                "fontSize": 18,
                "fontWeight": "normal",
              },
              {
                "color": "#FFFFFF",
              },
            ]
          }
          testID="com.ariesbifold:id/ToastBody"
        >
          Any success message
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "flex": 1,
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignSelf": "flex-end",
            "opacity": 1,
          }
        }
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            [
              {
                "color": "#FFFFFF",
                "fontSize": 24,
              },
              undefined,
              {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
        >
          
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`BaseToast Component Warn renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
>
  <View
    style={
      [
        {
          "alignItems": "center",
          "borderRadius": 4,
          "flex": 1,
          "flexDirection": "row",
          "marginTop": 60,
          "padding": 16,
        },
        {
          "backgroundColor": "#313132",
          "borderColor": "#FCBA19",
          "width": 675,
        },
      ]
    }
  >
    <View
      style={
        {
          "flex": 2,
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          {
            "flexShrink": 1,
          }
        }
      >
        <Text
          style={
            [
              {
                "color": "#FFFFFF",
                "fontSize": 18,
                "fontWeight": "normal",
              },
              {
                "fontSize": 14,
                "fontWeight": "bold",
              },
              {
                "color": "#FFFFFF",
              },
            ]
          }
          testID="com.ariesbifold:id/ToastTitle"
        >
          Any Warning Title
        </Text>
        <Text
          style={
            [
              {
                "color": "#FFFFFF",
                "fontSize": 18,
                "fontWeight": "normal",
              },
              {
                "color": "#FFFFFF",
              },
            ]
          }
          testID="com.ariesbifold:id/ToastBody"
        >
          Any warning message
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "flex": 1,
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignSelf": "flex-end",
            "opacity": 1,
          }
        }
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            [
              {
                "color": "#FFFFFF",
                "fontSize": 24,
              },
              undefined,
              {
                "fontFamily": "Material Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
        >
          
        </Text>
      </View>
    </View>
  </View>
</View>
`;
